### PR TITLE
Initial Commit of schema

### PR DESCRIPTION
**Pull request by rorysproule1**

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -0,0 +1,126 @@
+drop database if exists CareerLattice;
+create database CareerLattice;
+
+drop user if exists user@localhost;
+create user user@localhost identified with mysql_native_password by 'careerLattice!';
```
Comment (matthewdoreilly) [inline schema.sql]: Would be good to look into providing a hashed password here, so you don't have a plaintext password stored in GitHub.

File: schema.sql [hunk 2]
```diff
@@ -1,126 +1,133 @@
-drop database if exists CareerLattice;
-create database CareerLattice;
+DROP DATABASE IF EXISTS CareerLattice;
+CREATE DATABASE CareerLattice;
 
-drop user if exists user@localhost;
-create user user@localhost identified with mysql_native_password by 'careerLattice!';
+DROP USER IF EXISTS user@localhost;
+CREATE USER user@localhost IDENTIFIED WITH mysql_native_password BY 'careerLattice!';
```
Comment (jurassic-marc) [inline schema.sql]: Once this is hashed, looks all good to me

File: schema.sql [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+/** 
+CREATE VIEW viewRoles AS
+SELECT Department.DepartmentID, Department.Name AS 'Department Name', 
+Capability.CapabilityID, Capability.Name AS 'Capability Name', Band.BandID, Band.Name AS 'Band Name',
+Role.RoleID, Role.Name AS 'Role Name'
+FROM Department JOIN Capability ON Department.DepartmentID = Capability.DepartmentID JOIN Role_Capability ON
+Role_Capability.CapabilityID = Capability.CapabilityID JOIN Role ON Role_Capability.RoleID = Role.RoleID 
+JOIN Band ON Role.BandID = Band.BandID
+WHERE Department.Name = 'Sales & Marketing';
+**/
```
Comment (matthewdoreilly) [inline schema.sql]: I'd delete this, never include commented out code

File: schema.sql [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+CREATE VIEW viewLattice AS
+SELECT department.name AS 'Department Name', capability.name AS 'Capability Name', role.name AS 'Role Name'
+FROM department JOIN capability ON department.department_id = capability.department_id JOIN role_capability ON
+role_capability.capability_id = capability.capability_id JOIN role ON role_capability.role_id = role.role_id 
+WHERE department.department_id = 1;
+
+select * from viewLattice;
```
Comment (matthewdoreilly) [inline schema.sql]: What's this random SELECT statement doing? I'd remove this.

File: schema.sql [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+INSERT INTO band (name, school_id)
+VALUES ('Executive', 1), ('Leadership Executive', 1), ('Principal', 1), ('Manager', 2), ('Consultant', 2),
+('Senior Associate', 2), ('Associate', 1), ('Trainee', 1), ('Apprentice', 1);
+
+INSERT INTO role (name, department_id, band_id)
+VALUES ('Head of Business Unit', 1, 1), ('Head of Business Unit', 1, 1), ('Head of Business Unit', 1, 1), 
+('Head of Business Unit', 1, 1);
+INSERT INTO role (name, department_id, band_id)
+values ('Sales Director', 1, 2), ('Sales Director', 1, 2), ('Sales Director', 1, 2),
```
Comment (matthewdoreilly) [inline schema.sql]: Capitalise this

File: schema.sql [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+CREATE DATABASE career_lattice;
+USE career_lattice;
+
+-- ---------------------------------- TABLES ------------------------- --
+
+CREATE TABLE user (
+    user_id INT PRIMARY KEY AUTO_INCREMENT,
+    username VARCHAR(255) NOT NULL,
+    password VARCHAR(255) NOT NULL,
+    salt VARCHAR(255) NOT NULL,
```
Comment (matthewdoreilly) [inline schema.sql]: Is there a reason why there's a separate column for the salt? I would have thought the `password` column contains the plaintext password + salt, which is then all hashed together?

File: schema.sql [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+VALUES (27,3), (28,4), (29,5), (30,6);
+INSERT INTO role_capability (role_id, capability_id)
+VALUES (31,3), (32,4), (33,5), (34,6);
+INSERT INTO role_capability (role_id, capability_id)
+VALUES (35,4), (36,5), (37,6);
+
+-- ---------------------------------- VIEWS ------------------------------- --
+
+CREATE VIEW viewDepartments AS 
+SELECT department_id, Name
```
Comment (matthewdoreilly) [inline schema.sql]: `name` instead of `Name`

File: schema.sql [hunk 8]
```diff
@@ -1,128 +1,156 @@
...
+
+/** 
+CREATE VIEW viewRoles AS
+SELECT Department.DepartmentID, Department.Name AS 'Department Name', 
+Capability.CapabilityID, Capability.Name AS 'Capability Name', Band.BandID, Band.Name AS 'Band Name',
+Role.RoleID, Role.Name AS 'Role Name'
+FROM Department JOIN Capability ON Department.DepartmentID = Capability.DepartmentID JOIN Role_Capability ON
+Role_Capability.CapabilityID = Capability.CapabilityID JOIN Role ON Role_Capability.RoleID = Role.RoleID 
+JOIN Band ON Role.BandID = Band.BandID
+WHERE Department.Name = 'Sales & Marketing';
```
Comment (jamescrone) [inline schema.sql]: Having commented out code will be confusing to anyone pulling from master, if this code isn't needed, can it be removed?

**Pull request merged**